atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweepThread.java

```diff
@@ -203,40 +203,46 @@ SweepOutcome runOnce() {
     }
 
     private Optional<TableToSweep> getTableToSweep() {
-            return specificTableSweeper.getTxManager().runTaskWithRetry(
-                    tx -> {
-                        Optional<SweepProgress> progress = specificTableSweeper.getSweepProgressStore()
-                                .loadProgress(threadIndex);
-                        SweepPriorityOverrideConfig overrideConfig = sweepPriorityOverrideConfig.get();
-                        if (progress.map(
-                                realProgress -> shouldContinueSweepingCurrentTable(realProgress, overrideConfig))
-                                .orElse(false)) {
-                            try {
-                                createLockForTableIfNecessary(progress.get().tableRef());
-                                lockForCurrentTable.lockOrRefresh();
-                                return Optional.of(TableToSweep.continueSweeping(progress.get().tableRef(),
-                                        lockForCurrentTable,
-                                        progress.get()));
-                            } catch (InterruptedException ex) {
-                                log.info("Sweep lost the lock for table {}",
-                                        LoggingArgs.tableRef(progress.get().tableRef()));
-                                // We'll fall through and choose a new table
+        return specificTableSweeper.getTxManager().runTaskWithRetry(
+                tx -> {
+                    Optional<SweepProgress> progress = currentTable.flatMap(
+                            tableToSweep -> specificTableSweeper.getSweepProgressStore().loadProgress(
+                                    tableToSweep.getTableRef()));
+                    SweepPriorityOverrideConfig overrideConfig = sweepPriorityOverrideConfig.get();
+                    if (progress.map(
+                            realProgress -> shouldContinueSweepingCurrentTable(realProgress, overrideConfig))
+                            .orElse(false)) {
+                        try {
+                            createLockForTableIfNecessary(progress.get());
+                            if (currentTable.isPresent()) {
```

Comment:
The comment suppresses an intellij warning, which I guess occurs because potentially another thread can make currentTable empty in between isPresent and get, although in practice this won't happen here, because the class runs in a single thread.

Re the nit: I wish it could be, but that gives me a compile error (unhandled InterruptException) - yes, even though we catch it later.